src/tpm/tpm_wrapper.rs

```diff
@@ -178,63 +178,52 @@ pub fn ecdh(x: &[u8], y: &[u8], key_path: &str) -> Result<Vec<u8>> {
         let mut offset: tss2::size_t = 0;
         let c_path = CString::new(key_path.as_bytes()).unwrap();
 
-        let mut result = Fapi_GetEsysBlob(
+        tss2_call!(Fapi_GetEsysBlob(
             *tpm_ctx,
             c_path.as_ptr(),
             &mut blob_type as *mut u8,
             &mut esys_blob as *mut *mut u8,
             &mut blob_sz as *mut tss2::size_t,
-        );
-
-        if result != TSS2_RC_SUCCESS {
-            return Err(TpmError::tpm_error("Fapi_GetEsysBlob", result));
-        }
+        ))?;
 
         if blob_type != FAPI_ESYSBLOB_CONTEXTLOAD as u8 {
             return Err(TpmError::wrong_key_path());
         }
 
         let mut key_context: MaybeUninit<TPMS_CONTEXT> = MaybeUninit::uninit();
-        result = Tss2_MU_TPMS_CONTEXT_Unmarshal(
+        tss2_call!(Tss2_MU_TPMS_CONTEXT_Unmarshal(
             esys_blob,
             blob_sz,
             &mut offset as *mut tss2::size_t,
             key_context.as_mut_ptr(),
-        );
-        libc::free(esys_blob as *mut c_void);
-        if result != TSS2_RC_SUCCESS {
-            return Err(TpmError::tpm_error(
-                "Tss2_MU_TPMS_CONTEXT_Unmarshal",
-                result,
-            ));
-        }
+        ), always => {
+            libc::free(esys_blob as *mut c_void);
+        })?;
+
+        // let mut result;
```

Comment:
unneeded comment?

Reply:
yes, but somehow you're looking at an older commit? I'm not sure what happened. Perhaps you started a review long before submitting?